modules/_labs/core/react/lib/typeTest.ts

```diff
@@ -0,0 +1,139 @@
+import styled from '@emotion/styled';
+import {typeColors} from '@workday/canvas-colors-web';
+import {CSSProperties, fontFamily} from '@workday/canvas-kit-react-core';
+import {default as beta_Type} from './type';
+
+export interface CanvasTypeHierarchy {
+  level10: CSSProperties;
+  level9: CSSProperties;
+  level8: CSSProperties;
+  level7: CSSProperties;
+  level6: CSSProperties;
+  level5: CSSProperties;
+  level4: CSSProperties;
+  level3: CSSProperties;
+  level2: CSSProperties;
+  level1: CSSProperties;
+  [key: string]: CSSProperties;
+}
+
+const broswerDefaultFontSize = 16;
+/**
+ * Converts px values into base 16 rems
+ */
+const rem = (pxValue: number) => {
+  return `${pxValue / broswerDefaultFontSize}rem`;
+};
+
+/**
+ * The following rem values are based on an `html` element font size of 62.5%.
+ * This takes the browser default (16px) and converts it into base 10 at the default size.
+ * For example, 1.6rem == 16px.
+ */
+
+const hierarchy: CanvasTypeHierarchy = {
+  level10: {
+    fontFamily,
+    fontSize: rem(48),
+    lineHeight: rem(56),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level9: {
+    fontFamily,
+    fontSize: rem(40),
+    lineHeight: rem(48),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level8: {
+    fontFamily,
+    fontSize: rem(32),
+    lineHeight: rem(40),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level7: {
+    fontFamily,
+    fontSize: rem(24),
+    lineHeight: rem(32),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level6: {
+    fontFamily,
+    fontSize: rem(20),
+    lineHeight: rem(28),
+    letterSpacing: rem(0.16),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level5: {
+    fontFamily,
+    fontSize: rem(18),
+    lineHeight: rem(28),
+    letterSpacing: rem(0.16),
+    fontWeight: 700,
+    color: typeColors.heading,
+  },
+  level4: {
+    fontFamily,
+    fontSize: rem(16),
+    lineHeight: rem(24),
+    letterSpacing: rem(0.16),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+  level3: {
+    fontFamily,
+    fontSize: rem(14),
+    lineHeight: rem(20),
+    letterSpacing: rem(0.24),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+  level2: {
+    fontFamily,
+    fontSize: rem(12),
+    lineHeight: rem(16),
+    letterSpacing: rem(0.32),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+  level1: {
+    fontFamily,
+    fontSize: rem(10),
+    lineHeight: rem(16),
+    letterSpacing: rem(0.4),
+    fontWeight: 400,
+    color: typeColors.body,
+  },
+};
+
+type Variant = keyof typeof beta_Type.variant;
+type TextProps = {
+  level: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
```

Comment:
I really like the idea of `level`s. I think it helps users get it right. I wonder if `Text` should be limited to levels 1-4 and 5-10 are reserved for Headings to reduce the number of options. 🤔

Reply:
Also, any reason to not have these map to h1-h6 and text sizes?

Reply:
@alanbsmith there's are number of issues when it comes to mapping our typestyles to HTML semantic naming. From a systems view this isn't a scalable solution, too many type variables are built into these styles, limiting what we can offer. It also doesn't convey the properties of the type effectively.

In my mind we may also run into the same issue with Levels (if we are using levels from style naming that is)

Happy to catch up on this and discuss :)

Reply:
Thanks for the context, @seans-cummin. 😀

The main challenge for engineers with scalar props is learning to traverse the scale fluently. Most engineers tend to think in values, not scales (in my experience). Without an existing mental model it's challenging to keep it in your head if you don't think about type scales all the time. And the result can be significant frustration. If an engineer has to reference the documentation often to understand which level to use, it's not really an ergonomic win for them (they'll go back to styling it other ways). And that's fine, but there's no reason to keep the prop.

Some ways you can reduce the burden:
* reduce the number of steps
* make the baseline obvious 
* make a mid-scale baseline (if an engineer can find a mid-point, it's easier to move up or down)
* make the steps guessable
* use meaningful and / or memorable step names
* make step values easily discoverable

You don't need to use all of these, but I've seen combinations of them be effective for engineers to create mental models for type and spacing scales.

Reply:
Totally agree!

I'm in the process of formulating a workshop to inform how we should approach the naming convention from a design/engineering perspective so we can form a shared language when we talk about type.

Reply:
The more I think about `levels` the more I think they're really just Text variants.